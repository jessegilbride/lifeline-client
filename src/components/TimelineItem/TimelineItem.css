.TimelineItem {
  display: flex;
  /* flex-direction: column; */
  justify-content: flex-end;
  width: 50%;
  margin: 10px 0;
  padding-right: 1rem;
  text-decoration: none;
}

.TimelineItem:nth-child(odd) {
  align-self: flex-end;
  justify-content: flex-start;
  padding-left: 1rem;
  padding-right: unset;
}

.timeline-item-content {
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
    border-radius: 5px;
    background-color: #fff;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    padding: 15px;
    position: relative;
    width: 400px;
    max-width: 70%;
    text-align: right;
}

.timeline-item-content::after {
    content: ' ';
    background-color: #fff;
    box-shadow: 1px -1px 1px rgba(0, 0, 0, 0.2);
    position: absolute;
    right: -7px;
    top: calc(50% - 8px);
    transform: rotate(45deg);
    width: 15px;
    height: 15px;
}

.TimelineItem:nth-child(odd) .timeline-item-content {
  text-align: left;
  align-items: flex-start;
}

.TimelineItem:nth-child(odd) .timeline-item-content::after {
  right: auto;
  left: -7px;
  box-shadow: -1px 1px 1px rgba(0, 0, 0, 0.2);
}

.TimelineItem__heading {
  margin: 0;
}

