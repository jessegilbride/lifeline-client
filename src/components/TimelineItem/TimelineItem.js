import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { NiceDate } from '../Utils/Utils';
import './TimelineItem.css';

export default class TimelineItem extends Component {
  render() {
    
    // id, line_id, title, content, entry_date
    const { itemData } = this.props;
    
    return (
      <Link to={`/item/${itemData.id}`} className='TimelineItem'>
        <div className='timeline-item-content'>
          <header className='TimelineItem__header'>
            <h3 className='TimelineItem__heading'>{itemData.title}</h3>
          </header>
          <span className='TimelineItem__date'>
            <NiceDate date={itemData.entry_date} />
          </span>
        </div>
      </Link>
    );
  }
}

TimelineItem.defaultProps = {
  // id, line_id, title, content, entry_date
  itemData: {}
}
