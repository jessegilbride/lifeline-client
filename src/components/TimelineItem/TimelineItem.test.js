import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import TimelineItem from './TimelineItem';

describe('LineItem component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><TimelineItem /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><TimelineItem name="TimelineItem" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
