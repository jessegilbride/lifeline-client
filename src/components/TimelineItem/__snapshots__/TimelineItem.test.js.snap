// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItem component renders the UI as expected 1`] = `
<a
  className="TimelineItem"
  href="/item/undefined"
  onClick={[Function]}
>
  <div
    className="timeline-item-content"
  >
    <header
      className="TimelineItem__header"
    >
      <h3
        className="TimelineItem__heading"
      />
    </header>
    <span
      className="TimelineItem__date"
    >
      Invalid Date
    </span>
  </div>
</a>
`;
